bundle-workflow/src/test_workflow/integ/integ_test_suite.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import subprocess
+
+from git.git_repository import GitRepository
+from paths.script_finder import ScriptFinder
+from test_workflow.local_test_cluster import LocalTestCluster
+
+
+class IntegTestSuite:
+    """
+    Kicks of integration tests for a component based on test configurations provided in
+    test_support_matrix.yml
+    """
+    def __init__(self, component, test_config, bundle_manifest, work_dir):
+        self.component = component
+        self.bundle_manifest = bundle_manifest
+        self.work_dir = work_dir
+        self.test_config = test_config
+        self.script_finder = ScriptFinder()
+        self.repo = GitRepository(self.component.repository, self.component.commit_id, os.path.join(self.work_dir, self.component.name))
```

Comment:
This could use test_component and directly passed to integ_test_suite: https://github.com/opensearch-project/opensearch-build/blob/main/bundle-workflow/src/test_workflow/test_component.py